core/domain/stats_domain.py

```diff
@@ -1351,3 +1365,366 @@ def validate(self):
+                LearnerAnswerDetails.
+        """
+        self.state_reference = new_state_reference
+
+
+class LearnerAnswerInfo(object):
+    """Domain object containing the answer details submitted by the learner."""
+
+    def __init__(
+            self, learner_answer_info_id, answer,
+            answer_details, created_on):
+        """Constructs a LearnerAnswerInfo domain object.
+
+        Args:
+            learner_answer_info_id: str. The id of the LearnerAnswerInfo object.
+            answer: dict or str. The answer which is submitted by the learner.
+                Actually type of the answer is interaction dependent but mostly
+                it will be of dict type for most interactions, but it can be
+                string for TextInput interaction.
+            answer_details: str. The details the learner will submit when the
+                learner will be asked questions like 'Hey how did you land on
+                this answer', 'Why did you pick that answer' etc.
+            created_on: datetime. The time at which the answer details were
+                received.
+        """
+        self.id = learner_answer_info_id
+        self.answer = answer
+        self.answer_details = answer_details
+        self.created_on = created_on
+
+    def to_dict(self):
+        """Returns the dict of learner answer info.
+
+        Returns:
+            dict. The learner_answer_info dict.
+        """
+        learner_answer_info_dict = {
+            'id': self.id,
+            'answer': self.answer,
+            'answer_details': self.answer_details,
+            'created_on': str(self.created_on)
+        }
+        return learner_answer_info_dict
+
+    @classmethod
+    def from_dict(cls, learner_answer_info_dict):
+        """Returns a dict representing LearnerAnswerInfo domain object.
+
+        Returns:
+            dict. A dict, mapping all fields of LearnerAnswerInfo instance.
+        """
+
+        return cls(
+            learner_answer_info_dict['id'],
+            learner_answer_info_dict['answer'],
+            learner_answer_info_dict['answer_details'],
+            datetime.datetime.strptime(
+                learner_answer_info_dict['created_on'], '%Y-%m-%d %H:%M:%S.%f')
+        )
+
+    @classmethod
+    def get_learner_answer_info_id(cls, state_reference, entity_type):
+        """Generates the learner answer info domain object id.
+
+        Args:
+            state_reference: str. This is used to refer to a state
+                in an exploration or question. Like for an exploration the
+                value will be equal to 'exp_id.state_name' & for question
+                this will be equal to 'question_id' only.
+            entity_type: str. The type of entity i.e 'exploration' or
+                'question '.
+
+        Return:
+            learner_answer_info_id: str. The id generated by the function.
+        """
+        learner_answer_info_id = (state_reference + '.' + entity_type + '.' +
```

Comment:
Actually, this is the id of each learner answer info which is added in the list. So it needs to have random strings as there can be many learner answer info's, also it should be unique to identify it as it will help in deleting it.

Reply:
I see, ok. But in that case why do you need state reference and entity type in the ID? It just needs to be some randomly-generated string, right?

Reply:
Okay yeah, will correct it.

Reply:
Done